test/lib/ansible_test/_util/controller/sanity/validate-modules/validate_modules/schema.py

```diff
@@ -761,9 +763,15 @@ def return_schema(for_collection, plugin_type='module'):
 
 
 def deprecation_schema(for_collection):
+    alt_msg = 'Use either plural or singular, not both alternative(s)'
     main_fields = {
         Required('why'): doc_string,
-        Required('alternative'): doc_string,
+        Exclusive('plural', 'alt', msg=alt_msg): {
+            Required('alternatives'): doc_string
+        },
+        Exclusive('singular', 'alt', msg=alt_msg): {
+            Required('alternative'): doc_string
+        },
```

Comment:
The above does not work. It rejects both `alternatives` and `alternative` with errors like `extra keys not allowed @ data['deprecated']['alternative'].`. Basically doing what you want isn't easily supported with voluptuous (see https://github.com/alecthomas/voluptuous/issues/115), adding your own validator to make sure one of them is there is probably the best (similar to https://github.com/alecthomas/voluptuous/issues/115#issuecomment-144464666).

Try running this PR against a module that is deprecated (you can find some examples in community.general for example). (Would be good to have a test for that in ansible-core's integration tests for this as well.)

Reply:
We had very few deprecation message tests, the file above adds a bunch but i still had some 'todo' cases 
https://github.com/ansible/ansible/pull/82593/files#diff-a06276ae256ddf23396903d5813205f1ed799ab8e5f687c5adc87de6a4233863R37

Reply:
surprised this does not work, since it took it from the docs example
http://alecthomas.github.io/voluptuous/docs/_build/html/voluptuous.html#module-voluptuous.schema_builder

Reply:
I guess you refer to:
```py
msg = 'Please, use only one type of authentication at the same time.'
schema = Schema({
Exclusive('classic', 'auth', msg=msg):{
    Required('email'): str,
    Required('password'): str
    },
Exclusive('internal', 'auth', msg=msg):{
    Required('secret_key'): str
    },
Exclusive('social', 'auth', msg=msg):{
    Required('social_network'): str,
    Required('token'): str
    }
})
```
The example shows something else, as you can see from the code fragment which uses it:
```py
with raises(er.MultipleInvalid, "Please, use only one type of authentication at the same time. @ data[<auth>]"):
    schema({'classic': {'email': 'foo@example.com', 'password': 'bar'},
            'social': {'social_network': 'barfoo', 'token': 'tEMp'}})
```
There you can see that `classic` and `social` are sub-dictionaries, and `email`/`password`/... are not top-level keys.

Reply:
updated with custom function ... now need to figure out how to test it